### PR TITLE
Basic implementation for open subs

### PR DESCRIPTION
**Pull request by blazsolar**

### CODE REVIEW

File: provider/subs/opensubs/build.gradle [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+}
+
+repositories {
+    mavenCentral()
+    google()
+}
+
+dependencies {
+    implementation project(':provider:subs:provider')
+    api "io.reactivex.rxjava2:rxjava:$rxJavaVersion"
```
Comment (carlonzo) [inline provider/subs/opensubs/build.gradle]: you sure about `api` here?

File: provider/subs/opensubs/src/main/java/butter/droid/provider/subs/opensubs/OpenSubsProvider.java [hunk 2]
```diff
@@ -74,12 +79,38 @@ protected OpenSubsProvider(final OpenSubsService service, final Context context)
...
                 .map(SearchResponse::getData)
                 .flatMapObservable(Observable::fromIterable)
-//                .groupBy(OpenSubItem::getLanguageCode) // TODO group and rank
+                .groupBy(OpenSubItem::getLanguageCode)
+                .concatMap((Function<GroupedObservable<String, OpenSubItem>, ObservableSource<OpenSubItem>>) observable ->
+                        observable.reduce((openSubItem, openSubItem2) -> {
+                            int diff = getItemScore(openSubItem2) - getItemScore(openSubItem);
+                            // TODO downloads count
+//                            if (diff > 0 || diff == 0 && openSubItem2.getDownloads() > openSubItem.getDownloads()) {
+                            if (diff > 0 || diff == 0) {
```
Comment (carlonzo) [inline provider/subs/opensubs/src/main/java/butter/droid/provider/subs/opensubs/OpenSubsProvider.java]: you mean diff >= 0?

Comment (blazsolar) [reply provider/subs/opensubs/src/main/java/butter/droid/provider/subs/opensubs/OpenSubsProvider.java]: yes, will fix 👍

File: provider/subs/opensubs/src/main/java/butter/droid/provider/subs/opensubs/OpenSubsProvider.java [hunk 3]
```diff
@@ -85,7 +85,7 @@ protected OpenSubsProvider(final OpenSubsService service, final Context context)
                             int diff = getItemScore(openSubItem2) - getItemScore(openSubItem);
                             // TODO downloads count
 //                            if (diff > 0 || diff == 0 && openSubItem2.getDownloads() > openSubItem.getDownloads()) {
-                            if (diff > 0 || diff == 0) {
+                            if (diff > 0) {
```
Comment (carlonzo) [inline provider/subs/opensubs/src/main/java/butter/droid/provider/subs/opensubs/OpenSubsProvider.java]: `>=` ?

Comment (blazsolar) [reply provider/subs/opensubs/src/main/java/butter/droid/provider/subs/opensubs/OpenSubsProvider.java]: i think i got it now 😄

Comment (carlonzo) [reply provider/subs/opensubs/src/main/java/butter/droid/provider/subs/opensubs/OpenSubsProvider.java]: yeah that one! :D

File: provider/subs/opensubs/build.gradle (hunk 1 above)
Comment (blazsolar) [reply provider/subs/opensubs/build.gradle]: updated

**Pull request merged**